src/components/ProgramRecord/ProgramRecordActions.jsx

```diff
@@ -72,29 +72,34 @@ function ProgramRecordActions({
     }
   }, [downloadRecord]);
 
-  const handleProgramUrlCopy = () => {
-    if (programRecordUrl) {
-      navigator.clipboard.writeText(programRecordUrl);
-    } else {
-      getProgramRecordUrl(programUUID, username)
-        .then(({ data }) => {
-          setProgramRecordUrl(data.url);
-          navigator.clipboard.writeText(data.url);
-        })
-        .catch((error) => {
-          logError(error);
-          throw new Error(error);
-        });
-    }
+  const handleCopyEvent = () => {
     sendTrackEvent('edx.bi.credentials.program_record.share_url_copied', {
       category: 'records',
       'program-uuid': programUUID,
     });
+    setShowCopyTooltip(true);
   };
 
-  const handleCopyButtonClick = () => {
-    handleProgramUrlCopy();
-    setShowCopyTooltip(true);
+  const handleProgramUrlCopy = () => {
+    navigator.clipboard.writeText(programRecordUrl);
+    handleCopyEvent();
+  };
+
+  const handleProgramUrlCreate = () => {
+    getProgramRecordUrl(programUUID, username)
+      .then(({ data }) => {
+        setProgramRecordUrl(data.url);
+        navigator.clipboard.writeText(data.url);
+        sendTrackEvent('edx.bi.credentials.program_record.share_started', {
+          category: 'records',
+          'program-uuid': programUUID,
+        });
+      })
+      .catch((error) => {
+        logError(error);
+        throw new Error(error);
```

Comment:
Quick question; If we hit an error and we're here -- does it crash the entire page? Or would we render an alert to the learner?

I don't think we'd want to show some ugly error screen to the learner. If this is the case, can we open a follow-up ticket to fix this?

Reply:
Oh good point. We'd want to move the `handleCopyEvent` function into the `.then` and have some sort of "fail" notification to the user (e.g. a red tooltip that says "Link failed to copy"). Right now, it will show the blue tooltip "Link copied!" even if it fails. And I think it would copy the word "null" or some other false-y value. I can open a ticket for with these details under the LR MVP epic.